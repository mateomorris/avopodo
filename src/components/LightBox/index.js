import React, { Component } from 'react';
import { View, Text, Image, Dimensions, ScrollView, TouchableOpacity, Alert, Animated, Platform } from 'react-native';
import { Navigation } from "react-native-navigation";
import { BlurView } from 'react-native-blur';
import * as Animatable from 'react-native-animatable';
import { animate } from 'helpers/animations'

export class LightBox extends Component {

    state = {
        containerHeight : 0, // used to calculate how tall the touchable area under the content is
        viewHeight : 0, // this also
        opacity : new Animated.Value(0),
    }

    componentDidMount() {
        // Fade in
        animate([
            {
                property : this.state.opacity,
                toValue : 1,
                duration : 200,
            }
        ]);
    }

    _closeLightbox = () => {
        animate([
            {
                property : this.state.opacity,
                toValue : 0,
                duration : 200
            }
        ], () => {
            Navigation.dismissOverlay(this.props.componentId)
        });
    }


    render() {

        let window = Dimensions.get('window');

        return(
            <Animated.View style={{
                width: '100%',
                height: '100%',
                // backgroundColor: Platform.OS == 'ios' ?  'rgba(0,0,0,.5)' : 'rgba(0,0,0,.85)',
                padding: 10,
                // paddingTop: 37.5,
                paddingTop: 0,
                paddingBottom: 37.5,
                alignItems: 'center',
                opacity: this.state.opacity
            }}>
                <Image 
                    style={{ 
                        width: 25, 
                        height: 25,
                        position: 'absolute',
                        top: 35,
                        right: 20,
                        zIndex: 9
                    }} 
                    pointerEvents="none"
                    source={require('assets/x.png')} 
                />
                <BlurView
                    style={{
                        position: "absolute",
                        top: 0, left: 0, bottom: 0, right: 0,
                    }}
                    viewRef={this.state.viewRef}
                    blurType="dark"
                    blurAmount={10}
                />
                <View 
                style={{
                    maxHeight: window.height - 30,
                    paddingTop: 70
                }}
                onLayout={(e) => {
                    this.setState({
                        containerHeight : e.nativeEvent.layout.height
                    })
                }}
                >
                    {/* <TouchableOpacity style={{
                        paddingTop: 10,
                        paddingBottom: 10,
                        alignItems: 'flex-end'
                    }} onPress={() => {
                        this._closeLightbox()
                    }}>
                        <Image style={{ width: 25, height: 25 }} source={require('assets/x.png')} />
                    </TouchableOpacity> */}
                    <ScrollView style={[{
                        width: window.width - 40,
                        borderRadius: 5
                    }]} pointerEvents="box-none">
                        <View 
                        style={[{
                            backgroundColor: 'rgba(0,0,0,.5)',
                            height: 'auto',
                            width: window.width - 40,
                            borderRadius: 5
                        }, this.props.style]}
                        onLayout={(e) => {
                            this.setState({
                                viewHeight : e.nativeEvent.layout.height
                            })
                        }}
                        >
                            { this.props.children }
                        </View>
                        <View 
                            style={{
                                flex: 1,
                                height: this.state.containerHeight - this.state.viewHeight - 30
                            }}
                            onStartShouldSetResponder={() => {
                                this.background.setOpacityTo(0.2, 0.5);
                                return true
                            }}
                            onResponderRelease={() => {
                                this._closeLightbox()
                            }}
                            onResponderTerminate={() => {
                                this._closeLightbox()
                            }}
                            >
                        </View>
                    </ScrollView>
                    <TouchableOpacity 
                    ref={component => this.background = component}
                    style={{
                        position: 'absolute',
                        left: -50,
                        right: -50,
                        top: 0,
                        bottom: 0,
                        height: window.height,
                        zIndex: -1,
                        opacity: 1,
                        backgroundColor: 'rgba(0,0,0,.5)',
                    }} onPress={() => {
                        this._closeLightbox()
                    }}>
                    </TouchableOpacity>
                </View>
            </Animated.View>
        )
    }
}

LightBox.defaultProps = {
    style: {
        padding: 25
    }
}