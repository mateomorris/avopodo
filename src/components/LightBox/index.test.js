import React from 'react';
import { Text } from 'react-native';
import LightBox from 'components/LightBox';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<LightBox></LightBox>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});