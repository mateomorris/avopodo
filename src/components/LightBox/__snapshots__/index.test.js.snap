// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,.5)",
      "height": "100%",
      "opacity": 0,
      "padding": 10,
      "paddingBottom": 37.5,
      "paddingTop": 37.5,
      "width": "100%",
    }
  }
>
  <BlurView
    blurAmount={10}
    blurType="dark"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Object {
        "maxHeight": 1304,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/x.png",
          }
        }
        style={
          Object {
            "height": 25,
            "width": 25,
          }
        }
      />
    </View>
    <RCTScrollView
      style={
        Array [
          Object {
            "borderRadius": 5,
            "width": 710,
          },
        ]
      }
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,.5)",
                "borderRadius": 5,
                "height": "auto",
                "width": 710,
              },
              Object {
                "padding": 25,
              },
            ]
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "height": -30,
              "opacity": 1,
            }
          }
        />
      </View>
    </RCTScrollView>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 1334,
          "left": -50,
          "opacity": 1,
          "position": "absolute",
          "right": -50,
          "zIndex": -1,
        }
      }
    />
  </View>
</View>
`;
